core/services/job/job_orm_test.go

```diff
@@ -881,18 +848,37 @@ func Test_FindPipelineRunIDsByJobID(t *testing.T) {
 	for i, j := 0, 0; i < 2500; i++ {
 		mustInsertPipelineRun(t, pipelineORM, jobs[j])
 		j++
-		if j == len(jobs) {
+		if j == len(jobs)-1 {
 			j = 0
 		}
 	}
 
+	// Creation of job runs above cannot run in parallel, otherwise run ids are unpredictable
```

Comment:
Wait, couldn't this happen just as easily in prod though?

Reply:
No, this doesn't affect prod because the run id's don't have any significance.  Any number of unrelated run id's that show up in between run id's is fine.  The only reason we care about the run id's in this test is because we want to have precise control over how the batches work, in order to force an edge case where a problem with concatenating two batches together would be most likely to cause a problem.  These tests are verifying that even under extreme conditions, the concatenation doesn't affect the results.  This will all happen behind the scenes in prod.  Whether there are other processes in parallel affects how it gets batched but not the results.